jgrapht-core/src/main/java/org/jgrapht/intervalgraph/IntervalGraph.java

```diff
@@ -1,14 +1,468 @@
 package org.jgrapht.intervalgraph;
 
-import java.util.Collection;
-
+import org.jgrapht.EdgeFactory;
 import org.jgrapht.Graph;
+import org.jgrapht.GraphType;
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.*;
+import org.jgrapht.graph.specifics.FastLookupDirectedSpecifics;
+import org.jgrapht.graph.specifics.FastLookupUndirectedSpecifics;
+import org.jgrapht.graph.specifics.IntervalSpecifics;
+import org.jgrapht.graph.specifics.Specifics;
 import org.jgrapht.intervalgraph.interval.Interval;
+import org.jgrapht.intervalgraph.interval.IntervalVertex;
+import org.jgrapht.util.TypeUtil;
+
+import java.io.Serializable;
+import java.util.*;
+
+/**
+ * Implementation of an Interval Graph
+ *
+ * @param <V> the vertex type
+ * @param <E> the edge type
+ *
+ * @author Christoph Grüne (christophgruene)
+ * @since Apr 18, 2018
+ */
+public class IntervalGraph<V extends IntervalVertex, E> extends AbstractGraph<V, E> implements Graph<V, E>, Cloneable, Serializable {
+
+    private static final long serialVersionUID = 7835287075273098344L;
+
+    private static final String INTERVAL_GRAPH_ADD_EDGE = "Intervals of nodes define edges in interval graphs and cannot be modified manually";
+    private static final String GRAPH_SPECIFICS_MUST_NOT_BE_NULL = "Graph specifics must not be null";
+
+    private EdgeFactory<V, E> edgeFactory;
+    private transient Set<V> unmodifiableVertexSet = null;
+
+    private IntervalSpecifics<V, E> specifics;
+    private IntrusiveEdgesSpecifics<V, E> intrusiveEdgesSpecifics;
+
+    private boolean directed;
+    private boolean weighted;
+    private boolean allowingMultipleEdges;
+    private boolean allowingLoops;
```

Comment:
conventionally it will be better to rename them as isDirected, isWeighted, allowsMultipleEdges and allowsLoops.